src/analysis/matrix.lean

```diff
@@ -22,9 +33,20 @@ noncomputable theory
 
 open_locale nnreal
 
+namespace finset
+lemma prod_sup_mul_le_mul_sup_of_nonneg {ι κ α} [linear_ordered_semiring α] [order_bot α]
+  {a : ι → α} {b : κ → α} (s : finset ι) (t : finset κ)
+  (ha : ∀ i ∈ s, 0 ≤ a i) (hb : ∀ i ∈ t, 0 ≤ b i)  :
+  (s.product t).sup (λ p, a p.1 * b p.2) ≤ s.sup a * t.sup b :=
```

Comment:
@YaelDillies, do you think this is lemma worthy? Or should I be proving this inline?

Reply:
Yes, this lemma is worth having, but you should avoid `prod` in the name as the function is called `finset.product`!

Reply:
I've ended up removing it, the proof was easier without it.